src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int size;
+    private T[] arrayData;
+
+    public ArrayList() {
+        arrayData = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == arrayData.length) {
+            this.grow();
+        }
```

Comment:
It is better to move this check to the method
```suggestion
        grow();
```